watershed_workflow/condition.py

```diff
@@ -337,7 +337,7 @@ def __init__(self, cell, edges):
     assert (len(waterway.edges) == m2.num_edges)
 
     # delete the centroid info to force recalculation
-    m2.clear_geometry_cache()
+    # m2.clear_geometry_cache()
```

Comment:
@pinshuai  why did this get commented?  If the z-coordinates change, the centroid z-coordinate changes.  These should need to be recomputed?